mad-dog/src/tests/test_plays.js

```diff
@@ -0,0 +1,155 @@
+const axios = require('axios')
+const { delay } = require('../helpers.js')
+
+const MaxPollDurationMs = 240000
+
+async function logTrackListen(trackId, userId, solanaListen) {
+  return (await axios({
+    method: 'post',
+    url: `http://localhost:7000/tracks/${trackId}/listen`,
+    data: {
+      userId,
+      solanaListen,
+    }
+  })).data
+}
+
+async function getSolPlay (signature) {
+  return (await axios({
+    method: 'get',
+    url: `http://localhost:5000/get_sol_play?tx_sig=${signature}`
+  }))
+}
+
+async function getTotalPlays () {
+  return ((await axios({
+    method: 'get',
+    // use nonce to bypass cache
+    url: `http://localhost:5000/v1/metrics/plays?bucket_size=century&nonce=${Math.random()}`
+  })).data.data[0] || {}).count || 0
+}
+
+async function submitTrackListen (executeOne, trackId, userId, solanaListen) {
+  const initialPlays = getTotalPlays()
+
+  let signature
+  try {
+    await executeOne(0, async (libs) => {
+      const start = Date.now()
+      const identityResponse = await logTrackListen(trackId, userId, solanaListen)
+      signature = identityResponse.solTxSignature
+      console.log(`Logged track listen (trackId=${trackId}, userId=${userId}, solanaListen=${solanaListen}) | Processed in ${Date.now() - start}ms`)
+    })
+  } catch (err) {
+    console.log(`Failed to log track listen (trackId=${trackId}, userId=${userId}, solanaListen=${solanaListen}) with error ${err}`)
+    return false
+  }
+
+  const pollStart = Date.now()
+  console.log(`Polling track listen (trackId=${trackId}, userId=${userId}, solanaListen=${solanaListen})`)
+
+  if (solanaListen) {
+    let resp = (await getSolPlay(signature)).data
+    while (!resp.data) {
+      await delay(500)
+      resp = (await getSolPlay(signature)).data
+      if (Date.now() - pollStart > MaxPollDurationMs) {
+        throw new Error(`Failed to find ${signature} for userId=${userId}, trackId=${trackId} in ${MaxPollDurationMs}ms`)
+      }
+    }
+  } else {
+    let plays = await getTotalPlays()
+    while (plays === initialPlays) {
+      await delay(500)
+      plays = await getTotalPlays()
+      if (Date.now() - pollStart > MaxPollDurationMs) {
+        throw new Error(`Failed to find listen for userId=${userId}, trackId=${trackId} in ${MaxPollDurationMs}ms`)
+      }
+    }
+  }
+
+  console.log(`Found track listen (trackId=${trackId}, userId=${userId}, solanaListen=${solanaListen}) in discovery-provider`)
+
+  return true
+}
+
+async function trackListenCountsTest ({ executeOne }) {
+  const numBaseTrackListens = 10
+  const numAnonBaseTrackListens = 10
+  const numSolanaTrackListens = 10
```

Comment:
can we reflect the previous values here? it was 350 tracks w/between 2-5 listens per. down to drop the 2-5 listens but in that case let's record like 300 normal, 50 anon or something 

https://github.com/AudiusProject/audius-protocol/pull/1486/files#diff-6e733b33af7c083835bf79f53eaa309872d9d55412697d85bb9714f7b4362c2fL72